tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,338 @@
+    }
+
+##BRIDGE - USING AN EXISTING BRIDGE ON VM HOST SERVER
+##Prepare Test
+    my $vnet_host_bridge = <<'EOF';
+<network>
+<name>vnet_host_bridge</name>
+<forward mode="bridge"/>
+<bridge name="BRI"/>
+</network>
+EOF
+
+    #Create HOST BRIDGE NETWORK
+    assert_script_run("echo '$vnet_host_bridge' >> vnet_host_bridge.xml");
+    assert_script_run("sed -i -e 's/BRI/$virt_host_bridge/' vnet_host_bridge.xml");
+    assert_script_run("virsh net-create vnet_host_bridge.xml");
+    assert_script_run("virsh net-list --all|grep vnet_host_bridge");
+    save_screenshot;
+    upload_logs "vnet_host_bridge.xml";
+    assert_script_run("rm -rf vnet_host_bridge.xml");
+
+    my $gi_host_bridge = '';
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "HOST BRIDGE NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_host_bridge --live");
+        assert_script_run("ssh root\@$gi_guest 'rcnetwork restart'", 60);
+        if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
+            my $mac_host_bridge = script_output("virsh domiflist $guest|grep vnet_host_bridge|grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
+            script_retry "arp | grep $mac_host_bridge | awk \'{print \$1}\'", delay => 60, retry => 3, timeout => 60;
+            $gi_host_bridge = script_output("arp | grep $mac_host_bridge | awk \'{print \$1}\'");
+        }
+        else {
+            script_retry "virsh domifaddr $guest --source arp | grep vnet1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 3, timeout => 60;
+            $gi_host_bridge = script_output("virsh domifaddr $guest --source arp | grep vnet1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        }
+        #Destroy br123 test environment
+        my $mac_br123 = script_output("virsh domiflist $guest|grep br123|grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
+        assert_script_run("virsh detach-interface $guest bridge --mac $mac_br123");
+        destroy_standalone;
+        restart_libvirtd;
+        #Confirm HOST BRIDGE NETWORK
+        assert_script_run("ssh root\@$gi_host_bridge 'rcnetwork restart'",           60);
+        assert_script_run("ssh root\@$gi_host_bridge 'ping -c2 -W1 openqa.suse.de'", 60);
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest bridge --current");
+    }
+    #Destroy HOST BRIDGE NETWORK
+    assert_script_run("virsh net-destroy vnet_host_bridge");
+    save_screenshot;
+
+##NAT BASED NETWORK
+##Prepare Test
+    my $vnet_nated = <<'EOF';
+<network>
+  <name>vnet_nated</name>
+  <bridge name="virbr0"/>
+  <forward mode="nat"/>
+  <ip address="192.168.122.1" netmask="255.255.255.0">
+    <dhcp>
+      <range start="192.168.122.2" end="192.168.122.254"/>
+    </dhcp>
+  </ip>
+</network>
+EOF
+    die "There was not the default(NAT BASED NETWORK) virtual network" if (script_run('virsh net-list --all | grep default') != 0);
```

Comment:
Do you mean "The default(NAT BASED NETWORK) virtual network does not exist" ?

Reply:
Here, before NATed Network Test, just make sure "default" viral network existed on VM server.

Reply:
I think you should rephrase from my humble opinion.

Reply:
Great, had updated.

Reply:
Done